project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/execution/exception/KotlincExceptionTranslator.kt

```diff
@@ -38,6 +38,9 @@ class KotlincExceptionTranslator : BuildEventTranslator {
         }
 
     private fun transformCompilationError(buildEvent: BuildEvent) = either {
+        if (buildEvent.message.startsWith("[ksp]")) {
+            return@either parseKspError(buildEvent).bind()
+        }
         ensure(buildEvent is FileMessageEvent) {
```

Comment:
Does it fix the issue? 
`BuildSystemFail(cause=java.lang.IllegalStateException: Invalid event type: class com.intellij.build.events.impl.MessageEventImpl, but expected FileMessageEvent)`

Reply:
Well, this is not. Particularly, it cannot be solved in the simplest way. The compilation error should be represented using `(File, Line, Column)`. However, if that information is not available, that means that we have a translation error. So, I suppose to leave it this way to fix the one exception case by case